### ISSUE TITLE
Inverse & Root for KroneckerLazyVar

### ISSUE THREAD

**Issue by ssydasheng** [cornellius-gp/gpytorch#103]
For `KroneckerLazyVariable`, which you can easily get exact inverse and root, why you still go for conjugate gradient & lanczos ? These seems be slower to me cuz the true size of matrix is really big, by small steps of iterations, it might be far from a good solution.

**Comment by jacobrgardner** [comment]
The primary reason is that a `KroneckerLazyVariable` can represent the Kronecker product of other `LazyVariables`, which we might not want to evaluate or invert explicitly. This is true, for example, when doing KISS-GP with Kronecker structure on multiple dimensions (https://github.com/cornellius-gp/gpytorch/blob/master/examples/kissgp_kronecker_product_regression.ipynb).

It's true that if `KroneckerLazyVariable` were over only `NonLazyVariables`, we could probably avoid CG and Lanczos. 

Do you have an example use case where exploiting properties of the Kronecker product other than for MVMs would be useful? We just briefly discussed it and are open to the idea, we just can't think of what we'd gain, or how we'd keep the structure of the sub lazy variables intact.

**Comment by jacobrgardner** [comment]
It may also just generally be hard to justify using inversion algorithms other than CG at this point, even when they are available. CG typically gives fairly good solves with a very small number of iterations, making it a compelling choice even when an exact inverse is possible. For example, even for exact GPs, k iterations of CG requires O(kn^2) time compared to O(n^3) time to compute a Cholesky decomposition. Even without preconditioning, k can often be very small.

Furthermore, because in code we actually merge the CG and Lanczos iterations (see `gpytorch.utils.linear_cg`), the individual iterations use large matrix-matrix operations that utilize GPU resources well.

**Comment by ssydasheng** [comment]
Ah, I missed out `Multiplicative grid interpolation kernel` before. Thank you.
